mover-rsync/source.sh

```diff
@@ -39,19 +39,27 @@ Host *
   TCPKeepAlive no
 SSHCONFIG
 
+URL_DESTINATION_ADDRESS=$DESTINATION_ADDRESS
+if [[ "$DESTINATION_ADDRESS" == *"::"* || "$DESTINATION_ADDRESS" == *":"*":"*":"*":"*":"*":"*":"* ]]; then
+  echo "Destination address $DESTINATION_ADDRESS is ipv6"
+  if [[ "$DESTINATION_ADDRESS" != *"["*"]"* ]]; then
+    URL_DESTINATION_ADDRESS="[$DESTINATION_ADDRESS]"
+  fi
+fi
```

Comment:
The matching makes me nervous. What about bash regular expressions:
```bash
# Looking for either:
# 1) 8 groups of hex digits separated by ":"
# 2) a "::" in the string
IPV6_REGEX='(([0-9a-fA-F]{0,4}:){7}[0-9a-fA-F]{0,4})|(::)'

if [[ "$DESTINATION_ADDRESS" =~ $IPV6_REGEX ]]; then
    echo "It's IPv6"
fi

if [[ ! "$DESTINATION_ADDRESS" =~ \[.*\] ]]; then
    echo "no brackets"
fi
```

Reply:
That's much nicer than what I had - I'm terrible with regex's so threw my version together fast before I lost that ipv6 env...I've done a bunch of different address checks and I think your suggestion looks good to me - will update...